packages/zen-bull-netting/src/ZenBullNetting.sol

```diff
@@ -427,6 +484,143 @@ contract ZenBullNetting is Ownable, EIP712 {
         withdrawsIndex = i;
     }
 
+    /**
+     * @notice auction for queued withdraws
+     * @dev takes in orders from MM's to sell oSQTH
+     * @param _params Withdraw Params
+     */
+    function withdrawAuction(WithdrawAuctionParams calldata _params) public onlyOwner {
+        _checkOTCPrice(_params.clearingPrice, true);
+
+        uint256 initialEthBalance = address(this).balance;
+        uint256 bullTotalSupply = IERC20(zenBull).totalSupply();
+        uint256 crabAmount = _params.withdrawsToProcess * IZenBullStrategy(zenBull).getCrabBalance()
+            / bullTotalSupply;
+        (, uint256 crabDebt) = IZenBullStrategy(zenBull).getCrabVaultDetails();
+        uint256 oSqthAmount =
+            crabAmount * crabDebt / IERC20(IZenBullStrategy(zenBull).crab()).totalSupply();
+
+        // get oSQTH from market makers orders
+        uint256 toExchange = oSqthAmount;
+        for (uint256 i = 0; i < _params.orders.length && toExchange > 0; i++) {
+            _checkOrder(_params.orders[i]);
+            _useNonce(_params.orders[i].trader, _params.orders[i].nonce);
+
+            require(!_params.orders[i].isBuying, "ZBN19");
+            require(_params.orders[i].price <= _params.clearingPrice, "ZBN20");
+
+            if (_params.orders[i].quantity < toExchange) {
+                toExchange -= _params.orders[i].quantity;
+                IERC20(oSqth).transferFrom(
+                    _params.orders[i].trader, address(this), _params.orders[i].quantity
+                );
+            } else {
+                IERC20(oSqth).transferFrom(_params.orders[i].trader, address(this), toExchange);
+                break;
+            }
+        }
+
+        uint256 usdcToRepay = _params.withdrawsToProcess
+            * IEulerSimpleLens(eulerLens).getDTokenBalance(usdc, zenBull) / bullTotalSupply;
```

Comment:
Think this has different order of operations again (and not sure if any differences to use lens), so it might have a difference due to precision? 

contract calculates using this ` _bullShare.wmul(IEulerDToken(dToken).balanceOf(address(this)));`